src/components/AvatarMenu/AvatarMenu.tsx

```diff
@@ -0,0 +1,63 @@
+import {
+  Avatar,
+  IconButton,
+  Menu,
+  MenuItem,
+  Tooltip,
+  Typography,
+} from "@mui/material";
+import { useState } from "react";
+
+interface IAvatarMenuProps {
+  onLogoutClick: () => void;
+}
+const AvatarMenu = (props: IAvatarMenuProps) => {
+  const { onLogoutClick } = props;
+  const [anchorElUser, setAnchorElUser] = useState<null | HTMLElement>(null);
+
+  const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorElUser(event.currentTarget);
+  };
+
+  const handleCloseUserMenu = (setting: string) => {
```

Comment:
create a union type or use enum for settings

```suggestion
  const handleCloseUserMenu = (setting: SettingType) => () => {
```

Reply:
Fixed